### ISSUE TITLE
Error with code [attr-defined] when type ignoring the entire file

### ISSUE THREAD

**Issue by sh1ma** [python/mypy#9449]
Hello! I want judge if this is a bug.

Directory tree

```
helloworld
├── __init__.py
└── a.py
```

a.py
```python
# type: ignore


class Something:
    pass

```

\_\_init\_\_.py
```python
from .a import Something

```

mypy output
```
❯ mypy helloworld
helloworld/__init__.py:1: error: Module 'helloworld.a' has no attribute 'Something'
Found 1 error in 1 file (checked 2 source files)
```

However, when I removed `# type: ignore` on the first line of a.py

mypy output
```
❯ mypy helloworld
Success: no issues found in 2 source files
```

Is this a bug?

Thank you!

**Comment by gvanrossum** [comment]
No, it's as designed. The special marker `# type: ignore` at the top of a file means the file is not processed at all.

**Comment by dpinol** [comment]
Related to #9318
